modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -42,7 +42,8 @@
     protected String groupId = "io.swagger";
     protected String artifactId = "swagger-java-client";
     protected String artifactVersion = "1.0.0";
-    protected String sourceFolder = "src/main/java";
+    protected String projectFolder = "src/main";
+    protected String sourceFolder = projectFolder + "/java";
```

Comment:
But note that  sourceFolder is already "src/main/java"(without File.separator) in current java codegen